File: Sample_Code/C#.Net/Transaction_Notification.cs
/*
* BluePay C#.NET Sample code.
*
* This code sample shows a very based approach
* on handling data that is posted to a script running
* a merchant's server after a transaction is processed
* through their BluePay gateway account.
*/

using System;
using BPCSharp;
using System.Collections.Specialized;
using System.Web;
using System.Net;
using System.IO;
using System.Text;

namespace BP10Emu
{
    class Trans_Notify
    {
        public Trans_Notify()
        {
        }

        public static void Main()
        {

            HttpListener listener = new HttpListener();
            string secretKey = "";
            string response = "";

            try
            {
                // Listen for incoming data
                listener.Start();
            }
            catch (HttpListenerException)
            {
                return;
            }
            while (listener.IsListening)
            {
                var context = listener.GetContext();
                var body = new StreamReader(context.Request.InputStream).ReadToEnd();

                byte[] b = Encoding.UTF8.GetBytes("ACK");

                // Return HTTP Status of 200 to BluePay
                context.Response.StatusCode = 200;
                context.Response.KeepAlive = false;
                context.Response.ContentLength64 = b.Length;

                var output = context.Response.OutputStream;
                output.Write(b, 0, b.Length);

                // Get Reponse
                using (StreamReader reader = new StreamReader(output))
                {
                    response = reader.ReadToEnd();
                }
                context.Response.Close();
            }
            listener.Close();
            NameValueCollection vals = HttpUtility.ParseQueryString(response);

            // Parse data into a NVP collection
            string transID = vals["trans_id"];
            string transStatus = vals["trans_stats"];
            string transType = vals["trans_type"];
            string amount = vals["amount"];
            string batchID = vals["batch_id"];
            string batchStatus = vals["batch_status"];
            string totalCount = vals["total_count"];
            string totalAmount = vals["total_amount"];
            string batchUploadID = vals["batch_upload_id"];
            string rebillID = vals["rebill_id"];
            string rebillAmount = vals["rebill_amount"];
            string rebillStatus = vals["rebill_status"];

            // calculate the expected BP_STAMP
            string bpStamp = BluePayPayment_BP10Emu.calcTransNotifyTPS(secretKey,
                vals["trans_id"],
                vals["trans_stats"],
                vals["trans_type"],
                vals["amount"],
                vals["batch_id"],
                vals["batch_status"],
                vals["total_count"],
                vals["total_amount"],
                vals["batch_upload_id"],
                vals["rebill_id"],
                vals["rebill_amount"],
                vals["rebill_status"]);

            // Output data if the expected BP_STAMP matches the actual BP_STAMP
            if (bpStamp == vals["BP_STAMP"]) {
                Console.Write("Transaction ID: " + transID);
            Console.Write("Transaction Status: " + transStatus);
            Console.Write("Transaction Type: " + transType);
            Console.Write("Transaction Amount: " + amount);
            Console.Write("Rebill ID: " + rebillID);
            Console.Write("Rebill Amount: " + rebillAmount);
            Console.Write("Rebill Status: " + rebillStatus);
    } else {
                Console.Write("ERROR IN RECEIVING DATA FROM BLUEPAY");
    }
        }
    }
}

File: Sample_Code/Python/Transaction_Notification.py
##
     # BluePay Python Sample code.
     #
     # This code sample shows a very based approach
     # on handling data that is posted to a script running
     # a merchant's server after a transaction is processed
     # through their BluePay gateway account.
    ##
   
    from BluePayPayment_BP20Post import BluePayPayment_BP20Post
    import cgi
   
    vars = cgi.FieldStorage()
   
    secret_key = ""
    try:
        # Assign values
        transID = vars["trans_id"]
        transStatus = vars["trans_status"]
        transType = vars["trans_type"]
        amount = vars["amount"]
        batchID = vars["batch_id"]
        batchStatus = vars["batch_status"]
        totalCount = vars["total_count"]
        totalAmount = vars["total_amount"]
        batchUploadID = vars["batch_upload_id"]
        rebillID = vars["rebill_id"]
        rebillAmount = vars["reb_amount"]
        rebillStatus = vars["status"]
   
        # Calculate expected bp_stamp
        bp_stamp = BluePayPayment_BP20Post.calcTransNotifyTPS(secretKey,
            transID,
            transStatus,
            transType,
            amount,
            batchID,
            batchStatus,
            totalCount,
            totalAmount,
            batchUploadID,
            rebillID,
            rebillAmount,
            rebillStatus);
   
        # check if expected bp_stamp = actual bp_stamp
        if bp_stamp == vars["bp_stamp"]:
   
            # Get response from BluePay
            print 'Transaction ID: ' + trans_id
            print 'Transaction Status: ' + trans_status
            print  'Transaction Type: ' + trans_type
            print  'Transaction Amount: ' + amount
            print  'Rebill ID: ' + rebill_id
            print  'Rebill Amount: ' + rebill_amount
            print  'Rebill Status: ' + rebill_status
        else:
            print 'ERROR IN RECEIVING DATA FROM BLUEPAY'
    except KeyError:
        print "ERROR"